### ISSUE TITLE
Buildkit features trigger SC2215 for RUN flags that are technically valid statements

### ISSUE THREAD

**Issue by darkvertex** [hadolint/hadolint#347]
- [x] This is a bug report
- [X] This is a feature request (_Maybe?? Does Buildkit support count as a feature request?_)
- [X] I searched existing issues before opening this one

### Expected behavior

I would expect Hadolint to see that the 1st line in my file was:

```
# syntax=docker/dockerfile:experimental
```

...to see that it is indeed using *experimental* ([Buildkit](https://docs.docker.com/develop/develop-images/build_enhancements/)) features, so that when I do something like:

```Dockerfile
RUN --mount=type=ssh git clone --depth 1 -b master $REPO_ORIGIN_URL /path/to/my/repo
```

...it shouldn't complain about bad commands.

### Actual behavior

Hadolint gave me an `SC2215` error like this one:

```
This flag is used as a command name. Bad line break or missing [ .. ]? (SC2215)
```

### Steps to reproduce the behavior

NOTE: Buildkit is Linux-only and new in Docker as of version 18.09. *Ensure you are reproducing with this version or newer.*

1. Open a fresh terminal.
2. Make a temp folder and `cd` into it.
3. Drop the `Dockerfile` pasted below in there.
4. Ensure you have working SSH keys for Github for your own user (check with `ssh -T git@github.com`) and that you have an SSH "agent" running by running `echo $SSH_AUTH_SOCK` and if prints something, you do. If nothing prints, run `ssh-agent` and try again.
5. Run `export DOCKER_BUILDKIT=1` to enable Buildkit for docker build commands.
6. Run `docker build . --ssh default` to build with Buildkit's SSH forwarding feature enabled. It should forward your SSH agent for any `RUN --mount=type=ssh` lines and in theory it should print `You cloned successfully` if it worked.

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.17.1-0-gc123c8e
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:experimental
# (This build depends on BuildKit.)
FROM ubuntu:18.04
RUN --mount=type=ssh git clone git@github.com:hadolint/hadolint.git /root/hadolint && echo "You cloned successfully!"
```

**Additional environment details (OS, stack version, etc.)**

I'm on Docker version 19.03.1 (build 74b1e89) on Linux CentOS 7.3.1611 (though I don't think the OS matters for purposes of this issue.)

**Comment by adriantorrie** [comment]
To add to this:
- `# syntax = docker/dockerfile:1.0-experimental` (Note the version for the syntax)
- Buildkit is NOT Linux-only, it is available on Windows as well, someone else will have to confirm MacOS

**Comment by glensc** [comment]
Builtkit is definitely all platforms. I'm using docker for mac.

```
➔ echo 'RUN --mount=type=cache,id=apt,target=/var/cache/apt apt update' | docker run --rm -i hadolint/hadolint/dev/stdin:1 SC2215 This flag is used as a command name. Bad line break or missing [ .. ]?
➔
```

**Comment by jonpulsifer** [comment]
We are running into this using Hadolint in CI with GitHub Actions

**Comment by lorenzo** [comment]
I'm working on a new version of the parser to take care of the new flags. Should be done in a couple of weeks
